Integrations/integration-SentinelOne.yml

```diff
@@ -0,0 +1,2709 @@
+            'so-threat-summary': {
+                path: "threats/summary",
+                method: 'GET'
+            },
+            'so-mark-as-threat': {
+                path: 'threats/%threat_id%/mark-as-threat',
+                method: 'POST'
+            },
+            'so-mitigate-threat': {
+                path: 'threats/%threat_id%/mitigate/%action%',
+                method: 'POST'
+            },
+            'so-reslove-threats': {
+                path: 'threats/%threat_id%/resolve',
+                method: 'POST'
+            }
+        };
+
+        var queryMap = {
+            content_hash: true,
+            mitigation_status: true,
+            activity_type__in: true,
+            activity_type__nin: true,
+            resolved: true,
+            hidden: true,
+            display_name: true,
+            created_at__lt: true,
+            created_at__gt: true,
+            created_at__lte: true,
+            created_at__gte: true,
+            updated_at__lt: true,
+            updated_at__gt: true,
+            updated_at__lte: true,
+            updated_at__gte: true,
+            skip: true,
+            limit: true,
+            participating_fields: true,
+            query: true,
+            registered_at__lt: true,
+            registered_at__gt: true,
+            last_active_date__lt: true,
+            ast_active_date__gt: true,
+            total_memory__lte: true,
+            total_memory__gte: true,
+            core_count__lte: true,
+            core_count__gte: true,
+            cpu_count__lte: true,
+            cpu_count__gte: true,
+            is_active: true,
+            infected: true,
+            is_pending_uninstall: true,
+            is_decommissioned: true,
+            is_up_to_date: true,
+            encrypted_applications: true,
+            id__in: true,
+            id__nin: true,
+            group_id: true,
+            group_id__in: true,
+            policy_id__in: true,
+            computer_name__like: true,
+            agent_version__in: true,
+            os_type__in: true,
+            machine_type__in: true,
+            domain__in: true
+        };
+
+        var urlTemplate = "%url%/web/api/v%version%/";
+        var loginPath = 'users/login';
+        var logoutPath = 'users/logout';
+        var statusLBound = 200;
+        var statusUBound = 300;
+        var apiToken = params.token;
+        var userToken;
+        var fail = false;
+        var success = true;
+        var insecure = params.insecure;
+
+        var url = builder.url(urlTemplate, params);
+
+        function getqueryArgs(commadArgs) {
+            var queryArgs = {};
+            for (var arg in commadArgs) {
+                if (queryMap[arg]) {
+                    queryArgs[arg] = commadArgs[arg];
+                    delete commadArgs[arg];
+                }
+            return queryArgs;
+            }
+        }
+
+        var sendRequest = function(fullurl, method, headers, body) {
```

Comment:
we should take care of logout in case of failure and I am not sure we are doing that. please verify

Reply:
i will add it tonight

Reply:
@shaniacht1 Change category to `Endpoint` - One word